frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FieldRow/FieldRowContainer.tsx

```diff
@@ -310,50 +310,63 @@ export const FieldRowContainer = ({
 
 type MemoFieldRowProps = {
   field: FormFieldDto
-  questionNumber?: string
+  questionNumber?: number
 }
 
-const MemoFieldRow = memo(({ field, ...rest }: MemoFieldRowProps) => {
-  switch (field.fieldType) {
-    case BasicField.Attachment:
-      return <AttachmentField schema={field} {...rest} />
-    case BasicField.Checkbox:
-      return <CheckboxField schema={field} {...rest} />
-    case BasicField.Mobile:
-      return <MobileField schema={field} {...rest} />
-    case BasicField.HomeNo:
-      return <HomeNoField schema={field} {...rest} />
-    case BasicField.Email:
-      return <EmailField schema={field} {...rest} />
-    case BasicField.Nric:
-      return <NricField schema={field} {...rest} />
-    case BasicField.Number:
-      return <NumberField schema={field} {...rest} />
-    case BasicField.Decimal:
-      return <DecimalField schema={field} {...rest} />
-    case BasicField.Date:
-      return <DateField schema={field} {...rest} />
-    case BasicField.Dropdown:
-      return <DropdownField schema={field} {...rest} />
-    case BasicField.Statement:
-      return <ParagraphField schema={field} {...rest} />
-    case BasicField.ShortText:
-      return <ShortTextField schema={field} {...rest} />
-    case BasicField.LongText:
-      return <LongTextField schema={field} {...rest} />
-    case BasicField.Radio:
-      return <RadioField schema={field} {...rest} />
-    case BasicField.Rating:
-      return <RatingField schema={field} {...rest} />
-    case BasicField.Section:
-      return <SectionFieldRow field={field} {...rest} />
-    case BasicField.Uen:
-      return <UenField schema={field} {...rest} />
-    case BasicField.YesNo:
-      return <YesNoField schema={field} {...rest} />
-    case BasicField.Image:
-      return <ImageField schema={field} {...rest} />
-    case BasicField.Table:
-      return <TableField schema={field} {...rest} />
-  }
-})
+const MemoFieldRow = memo(
+  ({ field, questionNumber, ...rest }: MemoFieldRowProps) => {
+    // Q: The generation of question numbers is done way upstream and produces a
+    // number|undefined. Seems like there is no type guarantee that
+    // fieldType is submitted <==> questionNumber: number
+    // Is it worth refactoring to get this guarantee?
+
+    switch (field.fieldType) {
+      // Page fields with no question numbers
+      case BasicField.Section:
+        return <SectionFieldRow field={field} {...rest} />
+      case BasicField.Image:
+        return <ImageField schema={field} {...rest} />
+      case BasicField.Statement:
+        return <ParagraphField schema={field} {...rest} />
+      // User fields with question numbers
+      case BasicField.Attachment:
+        return (
+          <AttachmentField schema={{ ...field, questionNumber }} {...rest} />
```

Comment:
instead of recreating a new object with questionNumber in this `MemoFieldRow` component, should we pass in the field that is already augmented with question numbers using `frontend/src/features/form/utils/augmentWithQuestionNo.ts`?

traced the creation of the fields all the way up to `frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/useBuilderFields.ts`, that could be a good place to augment them.

Reply:
o yes you're right! hmm, `frontend/src/features/form/utils/augmentWithQuestionNo.ts` and `frontend/src/features/admin-form/create/builder-and-design/utils/questionNumbers.ts` both seem to be doing the same thing. Will get rid of the latter to clean up the code. Thanks for pointing that out!

Reply:
Note: this also takes care of the earlier comment about a possible improvement.